lighthouse-core/lib/manifest-parser.js

```diff
@@ -242,6 +242,18 @@ function parseIcon(raw, manifestUrl) {
 
   const type = parseString(raw.type, true);
 
+  const parsedPurpose = parseString(raw.purpose);
+  const purpose = {
+    raw: raw.purpose,
+    /** @type {string[]} */
```

Comment:
I don't *think* you need this now that it's `['any']` instead of `[]` but I could be wrong :)